README.md

```diff
@@ -1,2 +1,47 @@
 # secret-common-lib
-secret common lib to handle secret created on managed or un-managed k8s cluster on IBM Cloud
+secret common lib library handles authentication for access to ibmcloud resources by implementing secret providers.
+
+## secret providers
+
+The following interface can be seen defined in the secret-utils-lib library
+```
+// SecretProviderInterface ...
+type SecretProviderInterface interface {
+
+	// GetIAMToken ...
+	GetIAMToken(secret string, freshTokenRequired bool) (string, uint64, error)
+
+	// GetDefaultIAMToken ...
+	GetDefaultIAMToken(freshTokenRequired bool) (string, uint64, error)
+}
+```
+
+## Pre requisites
+- An environment variable IKS_ENABLED must to be set to true or false.
```

Comment:
It would be you explain what will happen `IKS_ENABLED` is set to true, also if its not set then the default is always false right?

Reply:
If it is not set, it is considered false.